### DISCUSSION TITLE
Floating model: release session via API

### DISCUSSION THREAD

**Discussion by r-brown** [Labs64/NetLicensing-Community#72] [Questions] [answered]
**Vendor request**

> We have some clients who only got a limited number of concurrent licenses to share between all their users. There has been a couple of times now where the license session has “hung” (probably due to the application crashing) so they have contacted us and we have gone in to the backend and deleted that hung session.
> My understanding is that the backend will time out an inactive session in 30 minutes. But is there a way that we can allow one of their admins to have permission to view and delete existing sessions through your API?
> So for example, user A is using the application but for some reason the license is not checked back in. User B wants to use the application but can’t because there isn’t a license available. They could wait 30 min for the backend to time out user As license. But is there a way that would allow their admin to get a list of active session ids and then delete them off? Hope that makes sense?

**Comment by v-rudkovskiy** [comment, marked as answer]
The easiest way to allow your customers to release such _"hung"_ sessions is to let them execute validation call for a given `sessionID` and `action=checkIn`.
Refer to the _"checkIn"_ [validation example](https://netlicensing.io/wiki/floating#validation).
Please note:
- this functionality might be misused by the end-customer, where the licenses will be checked-in often (on demand) to bypass [Floating model](https://netlicensing.io/wiki/floating) logic
- you need to use API Key role (`ROLE_APIKEY_LICENSEE`)
- we recommend the creation of additional API Key for a single customer; this way you can easily rewoke malicious API Keys
